treescript/src/treescript/merges.py

```diff
@@ -115,6 +115,28 @@ def touch_clobber_file(repo_path):
             "Merge old head via |hg debugsetparents {} {}| CLOSED TREE DONTBUILD a=release".format(tagged_from_rev, to_branch),
             repo_path=repo_path,
         )
+        # Preserve tags.
```

Comment:
I think this logic for generating the new `.hgtags` likely deserves its own function.

I also agree with the comment in gecko_migrations about thinking there should be a better way than generating patches. I think we should preserve the sentiment from that comment.